### ISSUE TITLE
feat: snarks js binding for gnark

### ISSUE THREAD

**Issue by smsunarto** [Consensys/gnark#917]
**Is your feature request related to a problem? Please describe.**

Gnark currently doesn't have a clean way to client side proving since snarksjs is reliant on a different r1cs encoding than the one gnark has.

**Describe the solution you'd like**
A snarkjs-gnark binding would allow snarksjs to do proving and witness generation for gnark circuits. 

Arkworks have something similar https://github.com/arkworks-rs/circom-compat

**Describe alternatives you've considered**
The hacky alternative is to compile gnark with tinygo, but other issues seems to have flagged performance and reliability issues.

**Comment by ivokub** [comment]
Yup, we have been thinking about having some internal IR which we can nicely transpile from/to other languages (circom, noir). Currently not a big priority though, for Linea we're using gnark end to end.

Imo I have seen some efforts now and then (I think there is Noir-to-gnark transpiler somewhere), but dunno how well maintained they are.
